File: 13_NoSQL/ExamsDb/Model/Student.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;

namespace ExamDbGenerator.Model
{
    class Student
    {
        public Student(StudentName name, Gender gender, Address address, DateOnly dateOfBirth, Class? currentClass = null)
        {
            Name = name;
            CurrentClass = currentClass;
            Gender = gender;
            Address = address;
            DateOfBirth = dateOfBirth;
        }

        [BsonId]
        public int Id => Name.Nr;
        public StudentName Name { get; set; }
        public Class? CurrentClass { get; set; }
        [BsonRepresentation(BsonType.String)]
        public Gender Gender { get; set; }
        public Address Address { get; set; }
        public DateOnly DateOfBirth { get; set; }
        [BsonElement]
        public List<Class> ClassHistory { get; } = new();
    }


}

File: 13_NoSQL/examsdb-java/src/main/java/at/spengergasse/examsdb/model/Student.java
package at.spengergasse.examsdb.model;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import org.bson.codecs.pojo.annotations.BsonId;

public record Student(StudentName name, Gender gender, Address address, LocalDate dateOfBirth, SchoolClass currentClass,
        List<SchoolClass> classHistory) {
    public Student {
    }

    public Student(StudentName name, Gender gender, Address address, LocalDate dateOfBirth, SchoolClass currentClass) {
        this(name, gender, address, dateOfBirth, currentClass, new ArrayList<SchoolClass>());
    }

    @BsonId
    public int id() {
        return name.nr();
    }
}